shared/gpuopen/src/posix/ddPosixSocket.cpp

```diff
@@ -202,56 +203,30 @@ namespace DevDriver
 
     Result Socket::Select(bool* pReadState, bool* pWriteState, bool* pExceptState, uint32 timeoutInMs)
     {
-        Result result = Result::Error;
-
-        fd_set readSet = {};
-        fd_set writeSet = {};
-        fd_set exceptSet = {};
-
-        FD_SET(m_osSocket, &readSet);
-        FD_SET(m_osSocket, &writeSet);
-        FD_SET(m_osSocket, &exceptSet);
+        Result result = Result::NotReady;
 
-        timeval timeoutValue = {};
-        timeoutValue.tv_sec = static_cast<int32>(timeoutInMs) / 1000;
-        timeoutValue.tv_usec = (static_cast<int32>(timeoutInMs) % 1000) * 1000;
+        pollfd socketPollFd;
+        socketPollFd.revents = 0;
+        socketPollFd.fd = m_osSocket;
+        socketPollFd.events = (((pReadState != nullptr) ? POLLIN : 0) | ((pWriteState != nullptr) ? POLLOUT : 0) | ((pExceptState != nullptr) ? POLLERR : 0));
```

Comment:
Can you make this multiple lines? This blows past our 120 char limit.

Reply:
Done